morphia/src/main/java/org/mongodb/morphia/AdvancedDatastore.java

```diff
@@ -247,6 +254,16 @@
     void setDecoderFact(DBDecoderFactory fact);
 
     /**
+     * Returns the {@see MongoCollection} for the given name
+     *
+     * @param <T>   the type of the MongoCollection
+     * @param name  the name of the collection
+     * @param clazz the type of the collection
+     * @return the collection
+     */
+    <T> MongoCollection<T> getMongoCollection(String name, Class<T> clazz);
```

Comment:
I see that it's used by the IndexHelper#createIndex methods.  How about the caller (which is just DatastoreImpl anyway) just passes in the MongoCollection (or the MongoDatabase) as an extra parameter to createIndex?  Then IndexHelper doesn't have to call back to Datastore just to get it.